src/pdb/mod.rs

```
@@ -234,31 +241,205 @@ impl PageFlags {
+    ///
+    /// In real databases, this has been found to be one of three things:
+    /// 1. The same value as `num_entries`.
+    /// 2. `0x1fff`. We assume this has the same meaning as **1.**
+    /// 3. A number smaller than `num_entries`, indicating the first empty
+    /// slot.
+    pub first_empty: u16,
+    /// The index entries.
+    #[br(count = num_entries)]
+    pub entries: Vec<IndexEntry>,
+}
+
+impl BinWrite for IndexPageContent {
+    type Args<'a> = (u32,);
+
+    fn write_options<W: Write + Seek>(
+        &self,
+        writer: &mut W,
+        endian: Endian,
+        (page_size,): Self::Args<'_>,
+    ) -> BinResult<()> {
+        let page_content_start_pos = writer.stream_position()?;
+
+        self.unknown_a.write_options(writer, endian, ())?;
+        self.unknown_b.write_options(writer, endian, ())?;
+        0x03ecu16.write_options(writer, endian, ())?;
+        self.next_offset.write_options(writer, endian, ())?;
+        self.page_index.write_options(writer, endian, ())?;
+        self.next_page.write_options(writer, endian, ())?;
+        0x0000_0000_03ff_ffffu64.write_options(writer, endian, ())?;
+        (self.entries.len() as u16).write_options(writer, endian, ())?;
+        self.first_empty.write_options(writer, endian, ())?;
+
+        for entry in &self.entries {
+            entry.write_options(writer, endian, ())?;
+        }
+
+        let after_entries_pos = writer.stream_position()?;
+        let written_bytes = after_entries_pos - page_content_start_pos;
+
+        let content_size = page_size - Page::HEADER_SIZE;
+        let padding_end_offset = content_size - 20;
+
+        /* Fill with empty entries (0x1ffffff8) until the last 20 bytes, which
+        are zeroes. If https://github.com/jam1garner/binrw/issues/205 was ever
+        fixed, this entire BinWrite implementation could possibly be removed.
+        */
+
+        if written_bytes < padding_end_offset as u64 {
```

Let's try to avoid `as` if `Into`/`From` also work. Reason is that if the types change so that this conversation is no longer lossless, the compiler will warn us about it. When using `as`, stuff would be silently lost. 

```suggestion
        if written_bytes < u64::from(padding_end_offset) {
```

---

yeah, I'd say avoid `as` in general if reasonable. It just silently truncates everything, which is seldom what you really want. https://doc.rust-lang.org/reference/expressions/operator-expr.html#r-expr.as.numeric